impeller/renderer/allocator.h

```diff
@@ -37,6 +37,15 @@ class Allocator {
   ///
   virtual uint16_t MinimumBytesPerRow(PixelFormat format) const;
 
+  // Debug/Profile only functionality for tracking the size of allocations
+#if (FLUTTER_RUNTIME_MODE != FLUTTER_RUNTIME_MODE_RELEASE)
+  const std::vector<ISize>& GetAllocatedSizes() const;
```

Comment:
I am not sure if you intend to [incorporate the allocator instrumentation into this patch](https://github.com/flutter/flutter/issues/120502#issuecomment-1426545766). If it helps speeds thing along for this patch, maybe we can back this out and land this patch while working on the instrumentation separately. Your call.

Reply:
I think now that I have an idea of how we will test this, I feel more comfortable backing this change out and working on it separately